### DISCUSSION TITLE
how do i hide an auto-increment field only inside thew edit & create view ??

### DISCUSSION THREAD

**Discussion by michabbb** [filamentphp/filament#5573] [Help] [answered]
i am new to filament and because everything is configured in the resource, I have no idea how to show different fields base on "view/edit/create" ;)

thanks for any help!

**Comment by ArtMin96** [comment, marked as answer]
I think
```php
->visible(static fn (Page $livewire): bool => $livewire instanceof ResourcePageClass)
```

**Reply by michabbb** [reply]
thanks. but i don´t see where you can differentiate between a view,edit,create ? you are constantly dealing with a "ResourceClass" but with different pages, that´s my problem.

**Reply by ArtMin96** [reply]
In the resource folder, you have a pages folder with create, edit, and view pages for resource. In the example I showed, write the name of your page instead of "ResourceClass".
`app/Filament/Resources/UserResource/Pages/CreateUser.php`

```php
->visible(static fn (Page $livewire): bool => $livewire instanceof CreateUser or EditUser or ...)
```

**Reply by michabbb** [reply]
thanks a lot for clarification 😏 it works, thanks a lot !!! 👍

**Reply by danharrin** [reply]
`hiddenOn()` / `visibleOn()` are cleaner - https://filamentphp.com/docs/2.x/admin/resources/getting-started#hiding-components-contextually
